test-tests/shared/src/test/scala/zio/test/environment/RandomSpec.scala

```diff
@@ -0,0 +1,168 @@
+package zio.test.environment
+
+import zio.test.Assertion._
+import zio.test.environment.RandomSpecUtil._
+import zio.test.environment.TestRandom.DefaultData
+import zio.test.{ assert, suite, test, testM, TestResult, ZIOBaseSpec }
+import zio._
+
+object RandomSpec
+    extends ZIOBaseSpec(
```

Comment:
You should be able to do:

```scala
object RandomSpec extends ZIOBaseSpec {
  // Helpers here

  suite(
    // Tests here
  )
}

Reply:
Thanks @jdegoes, that's actually what I tried to do in my first PR but unfortunately it doesn't compile
```
[info] Compiling 1 Scala source to C:\Users\rober\IdeaProjects\zio\test-tests\jvm\target\scala-2.12\test-classes ...
[error] C:\Users\rober\IdeaProjects\zio\test-tests\shared\src\test\scala\zio\test\environment\RandomSpec.scala:10:27: not enough arguments for constructor ZIOBaseSpec: (spec: => zio.test.ZSpec[zio.test.environment.TestEnvironment,Any,String,Any])zio.test.ZIOBaseSpec.
[error] Unspecified value parameter spec.
[error] object RandomSpec extends ZIOBaseSpec {
[error]                           ^
```
this is the approach that @adamgfraser suggested I take here https://github.com/zio/zio/pull/1959#discussion_r334951983
and also the approach shown here https://zio.dev/docs/usecases/usecases_testing
but if we can change `DefaultRunnableSpec` so this works that'd be excellent

I think the issue is with `RunnableSpec` - the class is abstract so doesn't need to define `spec` but I suspect it's been done so you don't have `override def spec` scattered throughout your tests. Perhaps @adamgfraser can shed some light on the background? 
```
abstract class RunnableSpec[R, L, T, E, S](runner0: TestRunner[R, L, T, E, S])(spec0: => Spec[R, E, L, T])
    extends AbstractRunnableSpec {
  ...
  override def spec   = spec0
```

Reply:
To use John's approach you need one more set of parentheses.

```scala
package zio.test.environment

import zio.test._
import zio.test.Assertion._

object RandomSpec
    extends ZIOBaseSpec({

      // Helpers here
      def plus(x: Int, y: Int): Int =
        x + y

      suite("RandomSpec")(
        // Tests here
        test("addition works fine") {
          assert(plus(1, 1), equalTo(2))
        }
      )
    })
```

You need the parentheses after `ZIOBaseSpec` (or `DefaultRunnableSpec`) to tell the compiler the code that follows is providing the required parameter to `ZIOBaseSpec` rather than the body of `RandomSpec` and then you need the brackets to be able to provide multiple statements. We did this to enable the user to not have to override anything but just be able to write tests in a declarative style.

```scala
objectMySpec extends DefaultRunnableSpec(
  suite("MySpec")(
    test("myTest")
```

It seems like so far users have preferred the separate object with helper methods style but both are fine.